File: esykart-store/esykart-store-merchant/src/main/resources/public/app/components/user/js/authenticationService.js
angular.module('altairApp').constant('url', 'rest/user/')
.service('AuthenticationService',
		['$http','$sessionStorage','utils','url',
		function($http,$sessionStorage, utils, url) {
			
//			mapping private function to public 
			var service = {};
	        service.authenticate = _authenticate;
	        return service;
			
//			private functions
	        function _authenticate(authenticationModel) {
	            return $http.post(url +"/authenticate",authenticationModel)
	            .then(utils.handleSuccess, utils.handleError('Error occurred while authenticating user.'));
	        }
		} 
]);

File: esykart-store/esykart-store-merchant/src/main/java/com/webientsoft/esykart/store/merchant/service/AuthenticationService.java
package com.webientsoft.esykart.store.merchant.service;

import com.webientsoft.esykart.common.model.common.AuthenticationModel;
import com.webientsoft.esykart.common.model.user.UserModel;

public interface AuthenticationService {
	
	UserModel authenticate(AuthenticationModel model);
	
	
}
